packages/@react-stately/table/src/useTableColumnResizeState.ts

```diff
@@ -0,0 +1,181 @@
+      }
+    }
+  }
+
+  function onColumnResizeStart() {
+    isResizing.current = true;
+    startResizeContentWidth.current = getContentWidth(columnWidthsRef.current);
+  }
+
+  function onColumnResize(column: GridNode<T>, width: number) {
+    let widthsObj = resizeColumn(column, width);
+    props.onColumnResize && props.onColumnResize(widthsObj);
+  }
+
+  function onColumnResizeEnd(column: GridNode<T>, width: number) {
+    isResizing.current = false;
+    if (props.onColumnResizeEnd) {
+      let widthsObj = resizeColumn(column, width);
+      props.onColumnResizeEnd(widthsObj);
+    }
+
+    let widths = new Map<Key, number>(columnWidthsRef.current);
+    widths.set(columnsRef.current[columnsRef.current.length - 1].key, 0);
+    setColumnWidthsForRef(widths);
+  }
+
+  function resizeColumn(column: GridNode<T>, newWidth: number) : { key: Key, width: number }[] {
+    // copy the columnWidths map and set the new width for the column being resized
+    let widths = new Map<Key, number>(columnWidthsRef.current);
+    widths.set(columnsRef.current[columnsRef.current.length - 1].key, 0);
+    widths.set(column.key, Math.max(
+      getMinWidth(column.props.minWidth, tableWidth.current),
+      Math.min(Math.floor(newWidth), getMaxWidth(column.props.maxWidth, tableWidth.current))
+    ));
+
+    // keep track of all columns that have been seized
+    resizedColumnsRef.current.add(column.key);
+    setResizedColumns(resizedColumnsRef.current);
+
+    // get the columns affected by resize and remaining space
+    const resizeIndex = columnsRef.current.findIndex(col => col.key === column.key);
+    let affectedColumns = columnsRef.current.slice(resizeIndex + 1);
+
+    // we only care about the columns that CAN be resized, we ignore static columns.
+    let {dynamicColumns} = getStaticAndDynamicColumns(affectedColumns);
+
+    // available space for affected columns
+    let availableSpace = columnsRef.current.reduce((acc, column, index) => {
+      if (index <= resizeIndex || isStatic(getRealColumnWidth(column))) {
+        return acc - widths.get(column.key);
+      }
+      return acc;
+    }, tableWidth.current);
+    
+    // merge the unaffected column widths and the recalculated column widths
+    let recalculatedColumnWidths = buildColumnWidths(dynamicColumns, availableSpace);
+    widths = new Map<Key, number>([...widths, ...recalculatedColumnWidths]);
+
+    if (startResizeContentWidth.current > tableWidth.current) {
+      widths.set(columnsRef.current[columnsRef.current.length - 1].key, Math.max(0, startResizeContentWidth.current - getContentWidth(widths)));
+    }
+    setColumnWidthsForRef(widths);
+
+    // when getting recalculated columns above, the column being resized is not considered "recalculated"
+    // so we need to add it to the list of affected columns
+    let allAffectedColumns = new Map<Key, number>([[column.key, newWidth], ...recalculatedColumnWidths]);
+    return mapToArray(allAffectedColumns);
```

Comment:
Can probably skip a couple steps here, in which case you can delete `mapToArray`

```suggestion
    return [[column.key, newWidth], ...recalculatedColumnWidths].map([key, width] => {key, width});
```

Reply:
oh can you spread maps like this into just a regular array? I'll have to test that out to make sure that works correctly.

Reply:
Haha, I was assuming so because that's technically what you are already doing.
But you can check it out working here too, open the console and enter this:

```
const map1 = new Map();
map1.set('a', 1);
map1.set('b', 2);
map1.set('c', 3);
console.log([...map1])
```

Reply:
I tested it and it did work! I made the change. I kept it named still just to make it clear what we're trying to get, but it definitely made it a bit more concise.